driver/src/orderbook/streamed/state.rs

```diff
@@ -66,7 +94,19 @@ impl State {
                     buy_amount,
                     sell_amount,
                 }
-            })
+            });
+        Ok(orders)
+    }
+
+    /// Returns whether the state needs to be updated with a pending solution that becomes the
+    /// accepted solution because the batch id has incremented. In this case
+    /// `apply_pending_solution_if_needed` has to be called before `account_state` and `orders` can
+    /// be used.
+    pub fn needs_to_apply_solution(&self, batch_id_: BatchId) -> bool {
+        match self.pending_solution {
+            PendingSolution::Submitted { batch_id, .. } if batch_id < batch_id_ => true,
+            _ => false,
```

Comment:
Can we distinguish the variable name a little bit more clearly please? One could be pending_solution_batch_id and the other one could be target_batch_id or just batch_id